api/src/pcapi/scripts/full_extract_offers.py

```diff
@@ -0,0 +1,111 @@
+import datetime
+import logging
+import statistics
+import time
+from typing import Iterable
+
+import click
+from flask import Blueprint
+import pytz
+from sqlalchemy.orm import joinedload
+
+import pcapi.core.offerers.models as offerers_models
+import pcapi.core.offers.models as offers_models
+from pcapi.core.search.backends import algolia
+from pcapi.core.users import utils
+
+
+# FIXME (apibrac, 2022-01-12): remove this file once the extract is done
+
+
+BATCH_SIZE = 1_000
+REPORT_EVERY = 10_000
+
+logger = logging.getLogger(__name__)
+blueprint = Blueprint(__name__, __name__)
+
+
+folder_name = "extracted_offers"
+
+
+def save_offers(operation_id, offers: Iterable[offers_models.Offer]) -> None:
+    serialized_offers = [str(algolia.AlgoliaBackend.serialize_offer(offer)) for offer in offers]
+    serialized_offers.append("")  # Adds an empty line at the end of the file. Essential for file composition.
+    file_content = "\n".join(serialized_offers)
```

Comment:
`algolia.AlgoliaBackend.serialize_offer()` renvoie un dictionnaire. Donc la fonction ici va exporter une représentation textuelle d'une liste de dictionnaire Python. Je ne sais pas si c'est très portable. On ne voudrait pas plutôt exporter en JSON ?

D'autre part, je ne comprends pas le commentaire "Essential for file composition" : tu as peut-être un cas d'utilisation que je n'ai pas en tête ? Dans mon idée, l'utilisateur récupérerait tous les fichiers et les importerait un par un (sans les concaténer au préalable).

Reply:
Pour le JSON : oui bien vu

La composition : il s'agit essentiellement de la commande `gsutil compose \` que j'ai mis dans la description de la pr : 
ça concatenne les fichiers et donc s'il n'y a pas de ligne vide à la fin, il va mettre sur la ligne du dernier élement la première offre du fichier suivant

En fait je comptais tout mettre en un seul gros fichier et l'envoyer comme ça.
Il ferrait 100Go tu pense que c'est trop gros à manipuler ?